pkg/secrets/secrets.go

```diff
@@ -23,6 +23,82 @@ var (
 	OverrideMountsFile = "/etc/containers/mounts.conf"
 )
 
+// secretData stores the name of the file and the content read from it
+type secretData struct {
+	name string
+	data []byte
+}
+
+// saveTo saves secret data to given directory
+func (s secretData) saveTo(dir string) error {
+	path := filepath.Join(dir, s.name)
+	if err := os.MkdirAll(filepath.Dir(path), 0700); err != nil && !os.IsExist(err) {
+		return err
+	}
+	return ioutil.WriteFile(path, s.data, 0700)
+}
+
+func readAll(root, prefix string) ([]secretData, error) {
+	path := filepath.Join(root, prefix)
+
+	data := []secretData{}
+
+	files, err := ioutil.ReadDir(path)
```

Comment:
would it make sense to use `ioutil.Walk`?  One advantage is that it also gives `os.FileInfo` without an additional Stat

Reply:
But does your patch walk the directories under the mounts.conf file?

IE If I have /usr/share/RHEL/secrets/foo/bar/is/data/in/this/dir

Will that show up in my container?

Reply:
yes, it walks the directories. It will check to see if "foo" is a difrectory, if it is it will open that up and read its files and so forth.